src/components/Dialogs/Dialog/Dialog.tsx

```diff
@@ -8,11 +8,16 @@ import styles from './Dialog.scss';
 export type ButtonProps = {
     onClick: () => void;
     label: string;
+    ButtonContext?: string; // this can also be set on the whole component instead..
 };
 
 export interface DialogProps extends ModalProps {
+    /** the component which will trigger the dialog */
+    dialogTrigger?: React.ReactNode;
     /** Should the alert appear */
     isOpen: boolean;
+    /** the function which triggers the modal */
```

Comment:
is this the callback that is executed after open/close of the dialog right? if yes, it might need some re write of the js doc